### PR TITLE
chore: use default dispatcher for non IO operations

### PR DESCRIPTION
**Pull request by ohassine**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

IO dispatcher is used for non IO operations.

### Solutions

Use Default dispatcher instead.

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: cryptography/src/commonJvmAndroid/kotlin/com.wire.kalium.cryptography/ProteusClientImpl.kt [hunk 1]
```diff
@@ -1,17 +1,26 @@
...
+    ioContext: CoroutineContext,
+    defaultContext: CoroutineContext
+) : ProteusClient {
 
     private var client: ProteusClient = (databaseKey?.let {
         ProteusClientCoreCryptoImpl(rootDir, it)
-    } ?: ProteusClientCryptoBoxImpl(rootDir)).also { kaliumLogger.d("Constructing ProteusClient #${instanceCounter++} at $rootDir") }
+    } ?: ProteusClientCryptoBoxImpl(rootDir, ioContext, defaultContext)).also {
+        kaliumLogger.d("Constructing ProteusClient #${instanceCounter++} at $rootDir")
+    }
```
Comment (MohamadJaara) [inline cryptography/src/commonJvmAndroid/kotlin/com.wire.kalium.cryptography/ProteusClientImpl.kt:23]: This log can be removed

File: cryptography/src/commonJvmAndroid/kotlin/com.wire.kalium.cryptography/ProteusClientImpl.kt [hunk 2]
```diff
@@ -1,17 +1,26 @@
 package com.wire.kalium.cryptography
 
+import kotlin.coroutines.CoroutineContext
+
 // TODO remove after debugging is completed
 var instanceCounter = 0
```
Comment (MohamadJaara) [inline cryptography/src/commonJvmAndroid/kotlin/com.wire.kalium.cryptography/ProteusClientImpl.kt:6]: same here

**Review by mchenani — APPROVED** [review]
🚨

**Comment by MohamadJaara** [comment]
we still need to limitedParallelism to 1 for all crypto box operations, and it will be easier to have it all run under one context, also function as tryGetSession under the hood is doing more that Io operations, so for simplicity  i would suggest to runit all under one context

**Pull request merged**